src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK013ThreadAffinitizedMEFConstruction.cs

```diff
@@ -0,0 +1,226 @@
+                {
+                    // Check if there is any other constructor decorated with importingConstructorAttribute
+                    if (containingType.Constructors.Any(ctor =>
+                        !SymbolEqualityComparer.Default.Equals(ctor, methodSymbol)
+                        && ctor.GetAttributes().Any(attr => SymbolEqualityComparer.Default.Equals(attr.AttributeClass, importingConstructorAttribute))))
+                    {
+                        // A different constructor is marked as importing contructor. It's OK if this constructor has UI thread dependency.
+                        return;
+                    }
+
+                    // Parameterless constructor must be free threaded
+                }
+                else if (methodSymbol.GetAttributes().Any(attr => SymbolEqualityComparer.Default.Equals(attr.AttributeClass, importingConstructorAttribute)))
+                {
+                    // Constructor decorated with ImportingConstructorAttribute must be free threaded
+                }
+                else if (partImportsSatisfiedNotificationInterface is not null
+                    && (methodSymbol.Name == Types.IPartImportsSatisfiedNotification.OnImportsSatisfied || methodSymbol.Name == Types.IPartImportsSatisfiedNotification.OnImportsSatisfiedFullName)
+                    && containingType.AllInterfaces.Contains(partImportsSatisfiedNotificationInterface, SymbolEqualityComparer.Default))
+                {
+                    // OnImportsSatisfied implementation must be free threaded
+                }
+                else
+                {
+                    // This analyzer does not enforce threading contstraints on any other methods.
+                    return;
+                }
+            }
+
+            ISymbol? targetSymbol = null;
+
+            // Use pattern matching to handle different operation types
+            switch (operation)
+            {
+                case IInvocationOperation invocationOperation:
+                    targetSymbol = invocationOperation.TargetMethod;
+                    break;
+
+                case IFieldReferenceOperation fieldReferenceOperation:
+                    targetSymbol = fieldReferenceOperation.Field;
+                    break;
+
+                case IPropertyReferenceOperation propertyReferenceOperation:
+                    targetSymbol = propertyReferenceOperation.Property;
+                    break;
+
+                case IMethodReferenceOperation methodReferenceOperation:
+                    targetSymbol = methodReferenceOperation.Method;
+                    break;
+
+                case IObjectCreationOperation objectCreationOperation:
+                    targetSymbol = objectCreationOperation.Constructor;
+                    break;
+
+                case IInstanceReferenceOperation instanceReferenceOperation:
+                    // not sure how to handle this
```

Comment:
I'd like help with this. I didn't see relevant data on the `IInstanceReferenceOperation` instance